### PR TITLE
Minor fixes for strange divergence bug. Note that this is only fixed …

### PR DESCRIPTION
**Pull request by JSKenyon**
…in the delay solver. Tagging @o-smirnov as this should fix the slope solver on your branch. I am leaving it up to you to resolve any other issues arising from the changes to restrict update.

### CODE REVIEW

File: cubical/machines/slope_machine.py [hunk 1]
```diff
@@ -255,18 +257,26 @@ def implement_update(self, jhr, jhjinv):
...
+        # out solutions are the parameters and not the gains themselves.
+        self._gh_update = self._ghinv_update = True
+
         if self.ref_ant is not None:
             # complicated slice :) we take the 0,0 phase offset of the reference antenna,
             # and subtract that from the phases of all other antennas and elements
-            slope_params -= slope_params[:,:,:,self.ref_ant,:,0,0][:,:,:,np.newaxis,:,np.newaxis,np.newaxis]
+            ref_slice = slice(self.ref_ant, self.ref_ant+1, 1)
+            ref_params = slope_params[:,:,:,ref_slice,:,(0,1),(0,1)]
+            slope_params[:,:,:,:,:,(0,1),(0,1)] -= ref_params
```
Comment (o-smirnov) [inline cubical/machines/slope_machine.py]: I'm not sure this is correct @JSKenyon -- I think only the 0,0 slope must be subtracted. In the presence of any polarized power, the RL or XY phase-zero difference is constrained by the off-diagonal data.

This does give me a most happy thought -- the PZD solution really belongs here and not in complex-2x2. It should be as easy as solving for two f-slopes.

Comment (JSKenyon) [reply cubical/machines/slope_machine.py]: Agreed @o-smirnov. Just pushed a minor fix to this branch to implement the correct reference behaviour.

**Comment by JSKenyon** [comment]
@o-smirnov Not entirely sure but it looks like there are some minor discrepancies between our branches now - not sure how that happened. Mostly just the minor fixes I suggested previously. That being said, this branch now includes my current effort on estimating delays from the FFT. The only added parameter is --(gain)-padding which allows you to specify the padding for the FFT. It defaults to zero. Padding improves the accuracy of the initial guess at the cost of increased memory footprint.

**Comment by KshitijT** [comment]
For the correct delay cal, should we be using issue-326 branch or the issue-326-jsk branch? Or neither?

**Comment by JSKenyon** [comment]
Either can work, but there is no guarantee that it is working properly yet - Oleg was still running into obstacles. I am away till early Jan at which point I will dig into it again.

**Comment by KshitijT** [comment]
Well, I get error when installing either of them:

```
    error: [Errno 2] No such file or directory: '/data/kshitij/CubiCal/cubical/bin/plot-bandpass-solutions'
```

Casa to the rescue, I guess. ;)

**Comment by o-smirnov** [comment]
I'm testing delay cal right now -- stay tuned. (I may have been testing an incorrect branch previously, so forget everything I said before...)

The plot-bandpass-solutions script needs to be removed from ``setup.py`` (I unified things into a single plot-gain-solutions script but forgot to update setup.py.) I'll do this on my branch for now, but you can do this yourself to unblock yourself.

**Pull request merged**